lightning-persister/src/lib.rs

```diff
@@ -130,7 +130,7 @@ impl FilesystemPersister {
 	}
 }
 
-impl<ChanSigner: ChannelKeys + Send + Sync> channelmonitor::Persist<ChanSigner> for FilesystemPersister {
+impl<ChanSigner: Sign + Send + Sync> channelmonitor::Persist<ChanSigner> for FilesystemPersister {
```

Comment:
Maybe here too?

Tangentially, I wonder if this type parameter should really be moved to the functions where it is clearer. Is it required at the struct level in order to work with another struct that is parameterized by this? Or is the compiler smart enough to make the determination?

Reply:
Hmm, this seems to not have any other Channel context.

As to the question about functions, I believe its required in this context because `ChannelMonitor` stores a signer in the struct, not just accepting it as a parameter to a function.

Reply:
For the context, I was seeing the module qualifier `channelmonitor` and the use below parameterizing `ChannelMonitor`. I don't feel too strongly about this, though.

Reply:
Ah. Yea, I see your point. I don't care either way now, and its more work to change it :).